### ISSUE TITLE
The first ARM (Apple Silicon) bottles are here, help us test them

### ISSUE THREAD

**Issue by fxcoudert** [Homebrew/brew#9488]
# Help needed

Our Apple Silicon CI is ready, hardware- and software-wise. We have built our first ARM bottles, and they are working fine on our testing machines. But before we go on building as many bottles as possible, we want to have more exposure of these bottles, and gather feedback. There are, necessarily, some kinks to work out in brew itself (and its bottling process) for ARM, since we're deploying to a new platform, and we want to make sure we identify as many as possible before we start mass-bottling.

**What can you do?** Test those new ARM bottles on your ARM Macs, and report below if you have any failure, problem, or other weirdness.

**How to report an issue?** Create a post below, and include:
- what you were doing (the command)
- what the output was (and give the full output in a link)
- **the output of `brew config`** in all cases (without it, for us it's like using a crystal ball)

**What bottles are available?**

```
autoconf automake bash bison boost
c-ares cairo cmake cscope fontconfig
freetype gdbm gettext git glib
gmp icu4c isl jpeg jsonlint
krb5 libevent libffi libomp libpng
libsodium libtiff libtool libyaml little-cms2
lua lzo meson mpfr mysql
ncurses ninja node openssl@1.1 p11-kit
pcre pcre2 perl pixman pkg-config
protobuf python@3.9 readline ruby sphinx-doc
sqlite tcl-tk vim xz zsh
```

List available with:

```
cd $(brew --repo homebrew/core)/Formula ; grep -l :arm64_big_sur *.rb | sed 's/.rb$//' | xargs
```

**Comment by unitof** [comment]
Do you want any confirmation that something works as expected, or only specific issues found?

**Comment by fxcoudert** [comment]
@unitof good question. Things _should_ be working, and I don't want to be flooded with “works for me too”-type posts, so let's go for: **please only report issues**/negative outcomes/problems with the bottles.

On the other hand, if things are working for you with one or more of these bottle, a 🚀  or ❤️ or other emoji response to this post will let me count if people have been actively testing, or not.

**Comment by deepj** [comment]
In the list of bottles for ARM is `node`. But when I tried to install `homebrew` started compiling `node`. Dependencies (`gdbm`, `xz`, `python@3.9`, `icu4c`) were installed without any compilation.

```
$ brew install node
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> Updated Formulae
Updated 1 formula.

Warning: You are running macOS on a arm64 CPU architecture.
We do not provide support for this (yet).
Reinstall Homebrew under Rosetta 2 until we support it.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
unsupported configuration.

==> Downloading https://homebrew.bintray.com/bottles/gdbm-1.18.1_1.arm64_big_sur.bottle.tar.gz
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/d5ae1727d7a669559f4f9333d165006ee5d2d7282a3e3a7aa1a28fddcdac6da2?response-content-disposition=attachment%3Bfilename%3D%22gdbm-1.18.1_1.arm64_big_
######################################################################## 100.0%
==> Downloading https://homebrew.bintray.com/bottles/xz-5.2.5.arm64_big_sur.bottle.tar.gz
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/409cface94d33051d3d5280af662e99c3e9d19a2a44d64e1d09a8af9717495d7?response-content-disposition=attachment%3Bfilename%3D%22xz-5.2.5.arm64_big_sur.b
######################################################################## 100.0%
==> Downloading https://homebrew.bintray.com/bottles/python%403.9-3.9.0_5.arm64_big_sur.bottle.tar.gz
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/5d890c25f640aa43b37102c7a922a723fa94886f8c8acb1e916eb267c297d526?response-content-disposition=attachment%3Bfilename%3D%22python%403.9-3.9.0_5.arm
######################################################################## 100.0%
==> Downloading https://homebrew.bintray.com/bottles/icu4c-67.1.arm64_big_sur.bottle.tar.gz
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/0d03423f2a4a557fc04a021fad963ab71d05b9129693887522dde996aff8c9f9?response-content-disposition=attachment%3Bfilename%3D%22icu4c-67.1.arm64_big_sur
######################################################################## 100.0%
==> Downloading https://registry.npmjs.org/npm/-/npm-7.0.15.tgz
######################################################################## 100.0%
==> Downloading https://nodejs.org/dist/v15.4.0/node-v15.4.0.tar.gz
######################################################################## 100.0%
==> Installing dependencies for node: gdbm, xz, python@3.9 and icu4c
==> Installing node dependency: gdbm
==> Pouring gdbm-1.18.1_1.arm64_big_sur.bottle.tar.gz
🍺  /opt/homebrew/Cellar/gdbm/1.18.1_1: 25 files, 876.2KB
==> Installing node dependency: xz
==> Pouring xz-5.2.5.arm64_big_sur.bottle.tar.gz
🍺  /opt/homebrew/Cellar/xz/5.2.5: 95 files, 1.4MB
==> Installing node dependency: python@3.9
==> Pouring python@3.9-3.9.0_5.arm64_big_sur.bottle.tar.gz
==> /opt/homebrew/Cellar/python@3.9/3.9.0_5/bin/python3 -s setup.py --no-user-cfg install --force --verbose --install-scripts=/opt/homebrew/Cellar/python@3.9/3.9.0_5/bin --install-lib=/opt/homebrew/lib/py
==> /opt/homebrew/Cellar/python@3.9/3.9.0_5/bin/python3 -s setup.py --no-user-cfg install --force --verbose --install-scripts=/opt/homebrew/Cellar/python@3.9/3.9.0_5/bin --install-lib=/opt/homebrew/lib/py
==> /opt/homebrew/Cellar/python@3.9/3.9.0_5/bin/python3 -s setup.py --no-user-cfg install --force --verbose --install-scripts=/opt/homebrew/Cellar/python@3.9/3.9.0_5/bin --install-lib=/opt/homebrew/lib/py
==> Caveats
Python has been installed as
  /opt/homebrew/bin/python3

Unversioned symlinks `python`, `python-config`, `pip` etc. pointing to
`python3`, `python3-config`, `pip3` etc., respectively, have been installed into
  /opt/homebrew/opt/python@3.9/libexec/bin

You can install Python packages with
  pip3 install <package>
They will install into the site-package directory
  /opt/homebrew/lib/python3.9/site-packages

See: https://docs.brew.sh/Homebrew-and-Python
==> Summary
🍺  /opt/homebrew/Cellar/python@3.9/3.9.0_5: 4,453 files, 72.8MB
==> Installing node dependency: icu4c
==> Pouring icu4c-67.1.arm64_big_sur.bottle.tar.gz
==> Caveats
icu4c is keg-only, which means it was not symlinked into /opt/homebrew,
because macOS provides libicucore.dylib (but nothing else).

If you need to have icu4c first in your PATH run:
  echo 'export PATH="/opt/homebrew/opt/icu4c/bin:$PATH"' >> ~/.zshrc
  echo 'export PATH="/opt/homebrew/opt/icu4c/sbin:$PATH"' >> ~/.zshrc

For compilers to find icu4c you may need to set:
  export LDFLAGS="-L/opt/homebrew/opt/icu4c/lib"
  export CPPFLAGS="-I/opt/homebrew/opt/icu4c/include"

For pkg-config to find icu4c you may need to set:
  export PKG_CONFIG_PATH="/opt/homebrew/opt/icu4c/lib/pkgconfig"

==> Summary
🍺  /opt/homebrew/Cellar/icu4c/67.1: 258 files, 72.4MB
==> Installing node
==> ./configure --prefix=/opt/homebrew/Cellar/node/15.4.0 --without-npm --with-intl=system-icu
==> make install
==> node /private/tmp/node-20201209-18177-7237f/node-v15.4.0/npm_bootstrap/bin/npm-cli.js install -ddd --global --prefix=/opt/homebrew/Cellar/node/15.4.0/libexec ~/Library/Caches/Homebrew/downl
🍺  /opt/homebrew/Cellar/node/15.4.0: 3,288 files, 54.6MB, built in 11 minutes 58 seconds
==> Caveats
==> python@3.9
Python has been installed as
  /opt/homebrew/bin/python3

Unversioned symlinks `python`, `python-config`, `pip` etc. pointing to
`python3`, `python3-config`, `pip3` etc., respectively, have been installed into
  /opt/homebrew/opt/python@3.9/libexec/bin

You can install Python packages with
  pip3 install <package>
They will install into the site-package directory
  /opt/homebrew/lib/python3.9/site-packages

See: https://docs.brew.sh/Homebrew-and-Python
==> icu4c
icu4c is keg-only, which means it was not symlinked into /opt/homebrew,
because macOS provides libicucore.dylib (but nothing else).

If you need to have icu4c first in your PATH run:
  echo 'export PATH="/opt/homebrew/opt/icu4c/bin:$PATH"' >> ~/.zshrc
  echo 'export PATH="/opt/homebrew/opt/icu4c/sbin:$PATH"' >> ~/.zshrc

For compilers to find icu4c you may need to set:
  export LDFLAGS="-L/opt/homebrew/opt/icu4c/lib"
  export CPPFLAGS="-I/opt/homebrew/opt/icu4c/include"

For pkg-config to find icu4c you may need to set:
  export PKG_CONFIG_PATH="/opt/homebrew/opt/icu4c/lib/pkgconfig"
```

So, `node` has a built bottle for ARM, or not?

**Comment by carlocab** [comment]
`node` has an arm bottle listed in the formula bottle block.

>How to report an issue? Create a post below, and include:
>- ...
>- the output of brew config in all cases (without it, for us it's like using a crystal ball)

**Comment by fxcoudert** [comment]
@deepj a bottle for node was built, but the update later today erased it. I am rebuilding, it should be available in ~15 minutes (`brew upgrade` should bring it to you)

**Comment by galak** [comment]
Is there a `brew` command that lists bottles?

**Comment by williammck** [comment]
> Is there a `brew` command that lists bottles?

@galak I put this together to list all bottles, part of the jq logic was from another issue in this repo. (sorry whoever you are, I don't remember 😦)

```
brew info --json=v1 --all | jq -r "map(select(.bottle.stable.files.arm64_big_sur) | .name) | unique | .[]"
```

It does take ~30s to run since it's loading all the formulae available, but it works. 🤷🏻

**Comment by carlocab** [comment]
Here's something that will take much than 30 seconds:
```
cd $(brew --repo homebrew/core)/Formula
rg -l ':arm64_big_sur'
```

If you don't like the `.rb` suffix,
```
cd $(brew --repo homebrew/core)/Formula
rg -l ':arm64_big_sur' | xargs -I{} basename {} .rb
```

**Comment by idyll** [comment]
python@3.9 didn't install from a bottle for me. (it did compile, just didn't come from the bottle.) literally everything else on that list worked though.

**Comment by carlocab** [comment]
Thanks for the report, @idyll. Could you please include the following information?

>How to report an issue? Create a post below, and include:
>
>- what you were doing (the command)
>- what the output was (and give the full output in a link)
>- the output of brew config in all cases (without it, for us it's like using a crystal ball)

**Comment by idyll** [comment]
- brew upgrade readline
- output: https://gist.github.com/idyll/5d0760dcb080275e664014a7ee8eb0ef
- ```
HOMEBREW_VERSION: 2.6.1-25-gfc921bb
ORIGIN: https://github.com/Homebrew/brew
HEAD: fc921bb640f1fc60eaa263db50fbeccf3fc66324
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 93697825da4944984a7b496f565fc01f4d0c63ad
Core tap last commit: 25 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vi
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.24.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.0.1-arm64
CLT: N/A
Xcode: 12.2
Rosetta 2: false
```

**Comment by carlocab** [comment]
This is why you don't get a bottle:
```
Warning: Building python@3.9 from source:
  The bottle needs the Apple Command Line Tools to be installed.
  You can install them, if desired, with:
    xcode-select --install
```
From your `brew config`:
```
...
CLT: N/A
...
```

**Comment by fxcoudert** [comment]
@idyll thanks for the report. The reason is stated in the output, our Python bottles need the command-line tools:

```
Warning: Building python@3.9 from source:
  The bottle needs the Apple Command Line Tools to be installed.
  You can install them, if desired, with:
    xcode-select --install
```

So it's expected behaviour that on your system (without CLT) they build from source.

**Comment by idyll** [comment]
Very strange. I am 99% sure I installed the CLT. But the CLT installation stuff has been so problematic with Big Sur that I think I've installed it 30 times throughout the course of the beta. 

And I just tried and it installed. So. 🤷

**Comment by tresf** [comment]
> Very strange. I am 99% sure I installed the CLT. But the CLT installation stuff has been so problematic with Big Sur that I think I've installed it 30 times throughout the course of the beta.
> 
> And I just tried and it installed. So. 🤷

In my experience, the Big Sur Beta tools often needed to be downloaded separately from the dev download portal.  When the M1 arrived, I avoided the Beta Channel for this single reason.

**Comment by hexiu** [comment]
brew install protobuf has error! 

==> Downloading https://files.pythonhosted.org/packages/6b/34/415834bfdafca3c5f451532e8a8d9ba89a21c97
Already downloaded: /Users/hejiahao/Library/Caches/Homebrew/downloads/e99698a932bde9f5fb8ca552923ae339984f284e1c7764df95c12175f573c0d7--six-1.15.0.tar.gz
==> Downloading https://github.com/protocolbuffers/protobuf/releases/download/v3.14.0/protobuf-all-3.
Already downloaded: /Users/hejiahao/Library/Caches/Homebrew/downloads/703fe5ff9c3437b572add06c8fb8f734731cf4650ea84d5052c934839a14313d--protobuf-all-3.14.0.tar.gz
==> ./configure --prefix=/opt/homebrew/Cellar/protobuf/3.14.0 --with-zlib
==> make

==> make check

==> make install
==> /opt/homebrew/opt/python@3.9/bin/python3 -c import setuptools... --no-user-cfg install --prefix=/
==> /opt/homebrew/opt/python@3.9/bin/python3 -c import setuptools... --no-user-cfg install --prefix=/
Last 15 lines from /Users/hejiahao/Library/Logs/Homebrew/protobuf/06.python3:
--prefix=/opt/homebrew/Cellar/protobuf/3.14.0/libexec
--install-scripts=/opt/homebrew/Cellar/protobuf/3.14.0/libexec/bin
--single-version-externally-managed
--record=installed.txt
--cpp_implementation

Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "setup.py", line 211, in <module>
    if mac_target and (pkg_resources.parse_version(mac_target) <
  File "/opt/homebrew/Cellar/python@3.9/3.9.0_3/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pkg_resources/__init__.py", line 113, in parse_version
    return packaging.version.Version(v)
  File "/opt/homebrew/Cellar/python@3.9/3.9.0_3/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pkg_resources/_vendor/packaging/version.py", line 275, in __init__
    match = self._regex.search(version)
TypeError: expected string or bytes-like object

Do not report this issue to Homebrew/brew or Homebrew/core!

**Comment by hexiu** [comment]
brew install protobuf@3.6

`==> /opt/homebrew/Cellar/sphinx-doc/3.3.1_1/libexec/bin/pip install -v --no-deps --no-binary :all: --
==> /opt/homebrew/Cellar/sphinx-doc/3.3.1_1/libexec/bin/pip install -v --no-deps --no-binary :all: --
==> /opt/homebrew/Cellar/sphinx-doc/3.3.1_1/libexec/bin/pip install -v --no-deps --no-binary :all: --
==> /opt/homebrew/Cellar/sphinx-doc/3.3.1_1/libexec/bin/pip install -v --no-deps --no-binary :all: --
==> /opt/homebrew/Cellar/sphinx-doc/3.3.1_1/libexec/bin/pip install -v --no-deps --no-binary :all: --
==> /opt/homebrew/Cellar/sphinx-doc/3.3.1_1/libexec/bin/pip install -v --no-deps --no-binary :all: --
Last 15 lines from /Users/hejiahao/Library/Logs/Homebrew/sphinx-doc/15.pip:
    requirement.install(
  File "/opt/Homebrew/Cellar/sphinx-doc/3.3.1_1/libexec/lib/python3.9/site-packages/pip/_internal/req/req_install.py", line 856, in install
    six.reraise(*exc.parent)
  File "/opt/Homebrew/Cellar/sphinx-doc/3.3.1_1/libexec/lib/python3.9/site-packages/pip/_vendor/six.py", line 703, in reraise
    raise value
  File "/opt/Homebrew/Cellar/sphinx-doc/3.3.1_1/libexec/lib/python3.9/site-packages/pip/_internal/operations/install/legacy.py", line 74, in install
    runner(
  File "/opt/Homebrew/Cellar/sphinx-doc/3.3.1_1/libexec/lib/python3.9/site-packages/pip/_internal/utils/subprocess.py", line 273, in runner
    call_subprocess(
  File "/opt/Homebrew/Cellar/sphinx-doc/3.3.1_1/libexec/lib/python3.9/site-packages/pip/_internal/utils/subprocess.py", line 242, in call_subprocess
    raise InstallationError(exc_msg)
pip._internal.exceptions.InstallationError: Command errored out with exit status 1: /opt/Homebrew/Cellar/sphinx-doc/3.3.1_1/libexec/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/tmp/pip-req-build-89sd1ogr/setup.py'"'"'; __file__='"'"'/private/tmp/pip-req-build-89sd1ogr/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/tmp/pip-record-sxwedzx3/install-record.txt --single-version-externally-managed --compile --install-headers /opt/Homebrew/Cellar/sphinx-doc/3.3.1_1/libexec/include/site/python3.9/MarkupSafe Check the logs for full command output.
WARNING: You are using pip version 20.2.4; however, version 20.3.1 is available.
You should consider upgrading via the '/opt/Homebrew/Cellar/sphinx-doc/3.3.1_1/libexec/bin/python -m pip install --upgrade pip' command.
Removed build tracker: '/private/tmp/pip-req-tracker-dfp0nn_m'

Do not report this issue to Homebrew/brew or Homebrew/core!
`

**Comment by hexiu** [comment]
brew config : 

HOMEBREW_VERSION: 2.6.0-51-gedb7496-dirty
ORIGIN: https://mirrors.ustc.edu.cn/brew.git
HEAD: edb7496630d3c6febab6003bf421eb072f1391bd
Last commit: 8 days ago
Core tap ORIGIN: https://mirrors.ustc.edu.cn/homebrew-core.git
Core tap HEAD: 88132eb4a1c8993a28c1a15b6ff2b1caf1b45f22
Core tap last commit: 8 days ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_REPOSITORY: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit dunno
Clang: 12.0 build 1200
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.0.1-arm64
CLT: 12.2.0.0.1.1604076827
Xcode: N/A
Rosetta 2: false

**Comment by ywwry66** [comment]
Your `brew` and `brew-core` are out of date

**Comment by carlocab** [comment]
>`Core tap HEAD: 88132eb4a1c8993a28c1a15b6ff2b1caf1b45f22`
>`Core tap last commit: 8 days ago`

This is your problem. There are no ARM bottles yet at this commit hash.

**Comment by ywwry66** [comment]
I tested with `git clone`, the mirror is up to date

**Comment by fxcoudert** [comment]
@hexiu thanks for the report. You need to `brew update` to make sure you get the ARM bottles. Also, note that `protobuf` should have a bottle, but `protobuf@3.6` is not yet supported.

Thanks to @carlocab and @ywwry66 for investigating

**Comment by fxcoudert** [comment]
@hexiu I see you have `CPU: octa-core 64-bit dunno` in your `brew config` output. Can you tell me what hardware (what exact mac model) you're using, and what the output of `sysctl hw.cpufamily` is for you?

**Comment by hexiu** [comment]
hw.cpufamily: 458787763    @fxcoudert

**Comment by carlocab** [comment]
@fxcoudert I think the `dunno` is explained by the outdated `brew`:
>ORIGIN: https://mirrors.ustc.edu.cn/brew.git
>HEAD: edb7496
>Last commit: 8 days ago

**Comment by hexiu** [comment]
@fxcoudert thanks，run `brew update`,  protobuf is installed,  protobuf@3.6 is error,  waiting python@3.9.1 merged https://github.com/Homebrew/homebrew-core/pull/66489

**Comment by fxcoudert** [comment]
@hexiu thanks!
@carlocab agreed, I wanted to double-check the value to see if we had missed something

**Comment by carlocab** [comment]
@hexiu do you still see `CPU: octa-core 64-bit dunno` in `brew config`?

**Comment by hexiu** [comment]
brew install protobuf@3.6

==> /opt/homebrew/opt/python@3.9/bin/python3 -c import setuptools... --no-user-cfg install --prefix=/
==> /opt/homebrew/opt/python@3.9/bin/python3 -c import setuptools... --no-user-cfg install --prefix=/
Last 15 lines from /Users/hejiahao/Library/Logs/Homebrew/protobuf@3.6/08.python3:
--prefix=/opt/homebrew/Cellar/protobuf@3.6/3.6.1.3_3/libexec
--install-scripts=/opt/homebrew/Cellar/protobuf@3.6/3.6.1.3_3/libexec/bin
--single-version-externally-managed
--record=installed.txt
--cpp_implementation

Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "setup.py", line 198, in <module>
    if mac_target and (pkg_resources.parse_version(mac_target) <
  File "/opt/homebrew/Cellar/python@3.9/3.9.1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pkg_resources/__init__.py", line 113, in parse_version
    return packaging.version.Version(v)
  File "/opt/homebrew/Cellar/python@3.9/3.9.1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pkg_resources/_vendor/packaging/version.py", line 275, in __init__
    match = self._regex.search(version)
TypeError: expected string or bytes-like object

Do not report this issue to Homebrew/brew or Homebrew/core!

**Comment by hexiu** [comment]
@carlocab 

➜  ~ abrew config
HOMEBREW_VERSION: 2.6.1-46-gd5283a7-dirty
ORIGIN: https://mirrors.ustc.edu.cn/brew.git
HEAD: d5283a77b95f1ce6740a429541f80bdad8a8da3b
Last commit: 11 hours ago
Core tap ORIGIN: https://mirrors.ustc.edu.cn/homebrew-core.git
Core tap HEAD: fff35e78c9b52392be323717075c78200063e60d
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.0.1-arm64
CLT: 12.2.0.0.1.1604076827
Xcode: N/A
Rosetta 2: false

**Comment by fxcoudert** [comment]
@hexiu protobuf@3.6 is not currently supported, only the main protobuf

**Comment by hexiu** [comment]
thanks,  Since the main protobuf has some new changes that are not compatible with the old GRPC, Let me update my own project @fxcoudert

**Comment by bdolewski** [comment]
Installing `swiftlint`:

- Command: `brew install --build-from-source swiftlint`
- Tested on my iOS projects, works very well
- Output: https://gist.github.com/bdolewski/8a6bdbff28edf9c861cb317bd4068e64
- config:
```
HOMEBREW_VERSION: 2.6.2-123-ga065dbc
ORIGIN: https://github.com/Homebrew/brew
HEAD: a065dbcc9b10b15af1708f430630e792ca7b187f
Last commit: 12 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: f089a94551a521192bc13ad75a1ed6d4219ae3da
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.29.2 => /opt/homebrew/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.1-arm64
CLT: 12.3.0.0.1.1607026830
Xcode: 12.3
Rosetta 2: false
```

**Comment by bdolewski** [comment]
I started to doubt if that's the Issue where I should put errors from packages being installed (compiled) from source rather than bottles created by Homebrew's CI. Please, redirect me if possible 🙂

Also tried to install `swiftgen`. Not so joyful results:

- Command: `brew install --build-from-source swiftgen`
- Output: https://gist.github.com/bdolewski/53359ba6379c7c8834939265bd9e7e8b
- Config:
```
➜  ~ brew config
HOMEBREW_VERSION: 2.6.2-123-ga065dbc
ORIGIN: https://github.com/Homebrew/brew
HEAD: a065dbcc9b10b15af1708f430630e792ca7b187f
Last commit: 13 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: fe40c5cb127c64b4a12aa47b0e94d35bf376a462
Core tap last commit: 7 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.29.2 => /opt/homebrew/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.1-arm64
CLT: 12.3.0.0.1.1607026830
Xcode: 12.3
Rosetta 2: false
```

- Error messages:
```
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/swiftgen/6.4.0/lib/Commander.framework/Versions/A/Commander
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/swiftgen/6.4.0/lib/Kanna.framework/Versions/A/Kanna
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/swiftgen/6.4.0/lib/PathKit.framework/Versions/A/PathKit
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/swiftgen/6.4.0/lib/Stencil.framework/Versions/A/Stencil
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/swiftgen/6.4.0/lib/StencilSwiftKit.framework/Versions/A/StencilSwiftKit
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/swiftgen/6.4.0/lib/SwiftGenKit.framework/Versions/A/SwiftGenKit
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/swiftgen/6.4.0/lib/Yams.framework/Versions/A/Yams
🍺  /opt/homebrew/Cellar/swiftgen/6.4.0: 203 files, 19.9MB, built in 1 minute 2 seconds
```

Are those issues related more to Homebrew or the `swiftgen` itself?

When I try to run `swiftgen` in Terminal, it got killed:
```
➜  ~ swiftgen
[1]    25847 killed     swiftgen
```

**Comment by bdolewski** [comment]
> python@3.9 didn't install from a bottle for me. (it did compile, just didn't come from the bottle.) literally everything else on that list worked though.
> 
> can I nominate Erlang 23.1.5 for a bottle next ;)

Can confirm as for today (19.12.2020). Cannot install *python@3.9* from bottle (ask me to compile it), the same with *vim*

**Comment by fxcoudert** [comment]
I'm trying to rebuild the Python bottle after an update, but there is a test issue:

```
==> /opt/homebrew/Cellar/python@3.9/3.9.1_1/bin/python3.9 -c import sqlite3
==> /opt/homebrew/Cellar/python@3.9/3.9.1_1/bin/python3.9 -c import tkinter; root = tkinter.Tk()
macOS 11 or later required !
An exception occurred within a child process:
```

Help welcome to investigate this

**Comment by carlocab** [comment]
Doesn't seem to be ARM specific, also happened on my Intel machine. Are you on Big Sur 11.1? @fxcoudert

**Comment by fxcoudert** [comment]
The backtrace says:

```
0   libsystem_kernel.dylib        	0x00007fff2031f462 __pthread_kill + 10
1   libsystem_pthread.dylib       	0x00007fff2034d610 pthread_kill + 263
2   libsystem_c.dylib             	0x00007fff202a0720 abort + 120
3   Tcl                           	0x00007fff6fe1eb55 Tcl_PanicVA + 398
4   Tcl                           	0x00007fff6fe1ebd5 Tcl_Panic + 128
5   Tk                            	0x00007fff6ff1ead5 TkpInit + 385
6   Tk                            	0x00007fff6fe9e788 0x7fff6fe6d000 + 202632
7   _tkinter.cpython-39-darwin.so 	0x0000000102d2d701 Tcl_AppInit + 84
8   _tkinter.cpython-39-darwin.so 	0x0000000102d289fa _tkinter_create + 975
9   org.python.python             	0x00000001027251bb cfunction_vectorcall_FASTCALL + 203
10  org.python.python             	0x000000010279cff3 call_function + 403
11  org.python.python             	0x000000010279a184 _PyEval_EvalFrameDefault + 27452
12  org.python.python             	0x000000010279db3b _PyEval_EvalCode + 1998
13  org.python.python             	0x00000001026f55d8 _PyFunction_Vectorcall + 248
14  org.python.python             	0x00000001026f4e67 _PyObject_FastCallDictTstate + 212
15  org.python.python             	0x00000001026f5891 _PyObject_Call_Prepend + 139
16  org.python.python             	0x000000010273de96 slot_tp_init + 87
17  org.python.python             	0x00000001027376f7 type_call + 150
18  org.python.python             	0x00000001026f4fa3 _PyObject_MakeTpCall + 266
19  org.python.python             	0x000000010279d027 call_function + 455
20  org.python.python             	0x000000010279a184 _PyEval_EvalFrameDefault + 27452
21  org.python.python             	0x000000010279db3b _PyEval_EvalCode + 1998
22  org.python.python             	0x000000010279356d PyEval_EvalCode + 79
23  org.python.python             	0x00000001027ce5b5 run_eval_code_obj + 110
24  org.python.python             	0x00000001027cd9ad run_mod + 103
25  org.python.python             	0x00000001027cc9dc PyRun_StringFlags + 182
26  org.python.python             	0x00000001027cc8e9 PyRun_SimpleStringFlags + 69
27  org.python.python             	0x00000001027e3802 Py_RunMain + 425
28  org.python.python             	0x00000001027e40c1 pymain_main + 306
29  org.python.python             	0x00000001027e410f Py_BytesMain + 42
```

**Comment by fxcoudert** [comment]
Regarding Python: there is an upstream bug report at https://bugs.python.org/issue42480
However I think it's an Apple Tcl/Tk issue, the same error occurs with Big Sur's python3

**Comment by fxcoudert** [comment]
ARM bottle for python@3.8 rebuilt

**Comment by SlowRoasted** [comment]
python@3.8 has been giving me the same error as python@3.9 had previously
`  File "./setup.py", line 971, in detect_readline_curses
    (tuple(int(n) for n in dep_target.split('.')[0:2])
AttributeError: 'int' object has no attribute 'split'`
I've tested applying the same fix that was done to 3.9 and it built successfully.

**Comment by deepj** [comment]
@fxcoudert Hello, is it possible to add more bottles for testing? I'd appreciate to try PostgreSQL, Redis, Erlang and Elixir.

**Comment by bdolewski** [comment]
I think `swiftlint` bottle can be added for testing. However I reported issue for SwiftGen team because, I tested 2 different build scenarios and it looks like their issue with builds in command line (building SwiftGen from Xcode works perfectly).

SwiftGen's issue:
https://github.com/SwiftGen/SwiftGen/issues/811

**Comment by fxcoudert** [comment]
So, a quick heads-up: we are readying a final (quick) round of testing before we start shipping ARM bottles in bigger numbers. Here is what is going to happen:

- the 50 existing bottles will be removed today
- they will be reintroduced in the next couple of hours
- I will post once that's done, and will kindly ask all of you to test the new bottles
- if we're satisfied that nothing's awry, we will start adding more bottles

Removing bottles… now!

**Comment by fxcoudert** [comment]
OK so we have 60 new bottles. **Please test those new bottles** and provide feedback (both positive and negative) below. You can test them by running `brew reinstall ninja && brew test ninja` (and similarly for other formulas).

I intend to leave about 24 hours to see if something is broken, and if not, start bottling at a faster pace.

```
autoconf automake bash bison c-ares cairo cmake cscope
fontconfig freetype gdbm gettext git glib gmp gnu-sed
icu4c isl jemalloc jpeg krb5 libev libevent libffi
libidn libmpc libomp libpng libsodium libtasn1 libtiff libtool
libunistring libyaml little-cms2 lua lzo meson mpfr ncurses
nghttp2 ninja node openssl@1.1 p11-kit pcre pcre2 perl
pixman pkg-config protobuf python@3.9 readline redis ruby
sphinx-doc sqlite tcl-tk xz zsh
```

Thanks again to everyone here helping us test on ARM!

**Comment by bdolewski** [comment]
I ran `brew reinstall` and `brew test` on a bunch of bottles:
```
git
gdbm
openssl@1.1
readline
sqlite
xz
python
libyaml
```

No issues for those whatsoever. However I tried the same steps for ruby with following results:
```
➜  ~ brew reinstall ruby
Warning: You are running macOS on a arm64 CPU architecture.
We do not provide support for this (yet).
Reinstall Homebrew under Rosetta 2 until we support it.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
unsupported configuration.

==> Downloading https://homebrew.bintray.com/bottles/ruby-2.7.2.arm64_big_sur.bottle.tar.gz
Already downloaded: /Users/bartoszdolewski/Library/Caches/Homebrew/downloads/1a670e50d3cc3354f8d08e1d33bc3557d55f01b6dad7ae231ac4a0340f62a8b2--ruby-2.7.2.arm64_big_sur.bottle.tar.gz
==> Reinstalling ruby 
==> Pouring ruby-2.7.2.arm64_big_sur.bottle.tar.gz
==> Caveats
By default, binaries installed by gem will be placed into:
  /opt/homebrew/lib/ruby/gems/2.7.0/bin

You may want to add this to your PATH.

ruby is keg-only, which means it was not symlinked into /opt/homebrew,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

If you need to have ruby first in your PATH run:
  echo 'export PATH="/opt/homebrew/opt/ruby/bin:$PATH"' >> ~/.zshrc

For compilers to find ruby you may need to set:
  export LDFLAGS="-L/opt/homebrew/opt/ruby/lib"
  export CPPFLAGS="-I/opt/homebrew/opt/ruby/include"

==> Summary
🍺  /opt/homebrew/Cellar/ruby/2.7.2: 20,156 files, 37.2MB
➜  ~ brew test ruby
==> Testing ruby
==> /opt/homebrew/Cellar/ruby/2.7.2/bin/ruby -e 'puts :hello'
==> /opt/homebrew/Cellar/ruby/2.7.2/bin/gem install json
==> /opt/homebrew/Cellar/ruby/2.7.2/bin/bundle exec ls
Last 15 lines from /Users/bartoszdolewski/Library/Logs/Homebrew/ruby/test.02.bundle:
2020-12-21 19:03:51 +0100

/opt/homebrew/Cellar/ruby/2.7.2/bin/bundle
exec
ls

/opt/homebrew/Cellar/ruby/2.7.2/bin/bundle:23:in `load': cannot load such file -- /opt/homebrew/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/exe/bundle (LoadError)
	from /opt/homebrew/Cellar/ruby/2.7.2/bin/bundle:23:in `<main>'
Error: ruby: failed
An exception occurred within a child process:
  BuildError: Failed executing: /opt/homebrew/Cellar/ruby/2.7.2/bin/bundle exec ls
/opt/homebrew/Library/Homebrew/formula.rb:2072:in `block in system'
/opt/homebrew/Library/Homebrew/formula.rb:2008:in `open'
/opt/homebrew/Library/Homebrew/formula.rb:2008:in `system'
/opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/ruby.rb:240:in `block in <class:Ruby>'
/opt/homebrew/Library/Homebrew/formula.rb:1876:in `block (3 levels) in run_test'
/opt/homebrew/Library/Homebrew/utils.rb:530:in `with_env'
/opt/homebrew/Library/Homebrew/formula.rb:1875:in `block (2 levels) in run_test'
/opt/homebrew/Library/Homebrew/formula.rb:904:in `with_logging'
/opt/homebrew/Library/Homebrew/formula.rb:1874:in `block in run_test'
/opt/homebrew/Library/Homebrew/mktemp.rb:63:in `block in run'
/opt/homebrew/Library/Homebrew/mktemp.rb:63:in `chdir'
/opt/homebrew/Library/Homebrew/mktemp.rb:63:in `run'
/opt/homebrew/Library/Homebrew/formula.rb:2121:in `mktemp'
/opt/homebrew/Library/Homebrew/formula.rb:1868:in `run_test'
/opt/homebrew/Library/Homebrew/test.rb:43:in `block in <main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:93:in `block in timeout'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `block in catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:108:in `timeout'
/opt/homebrew/Library/Homebrew/test.rb:42:in `<main>'
```

**Comment by fxcoudert** [comment]
@bdolewski thanks, can you edit your post and paste the output of `brew config` and `brew doctor` please? I'll look into it

**Comment by bdolewski** [comment]
No problem :)

**brew config**
```
➜  ~ brew config
HOMEBREW_VERSION: 2.7.0-2-g6994633
ORIGIN: https://github.com/Homebrew/brew
HEAD: 699463358167072d577d0f13f76b0722e32d8cc0
Last commit: 3 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 84c6c43b31ece5cc412db7092a2331d9a843d753
Core tap last commit: 27 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.29.2 => /opt/homebrew/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.1-arm64
CLT: 12.3.0.0.1.1607026830
Xcode: 12.3
Rosetta 2: false
```

**brew doctor**
```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You are running macOS on a arm64 CPU architecture.
We do not provide support for this (yet).
Reinstall Homebrew under Rosetta 2 until we support it.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
unsupported configuration.
```

**uname -a** (bonus input ;) )
```
Darwin MacBook-Pro-Bartosz.local 20.2.0 Darwin Kernel Version 20.2.0: Wed Dec  2 20:40:21 PST 2020; root:xnu-7195.60.75~1/RELEASE_ARM64_T8101 arm64
```

**Comment by scottstanfield** [comment]
Success! Installed from bottle and tested

```
bash readline git openssl@1.1 python@3.9 ruby zsh autoconf automake cmake lua sqlite
```

**Comment by bdolewski** [comment]
@scottstanfield did you run `brew test ruby` with success? I did `rm -rf $(brew --cache)` and then `brew instal ruby`, `brew test ruby` again. The same failure as before 🤨

**Comment by MikeMcQuaid** [comment]
@bdolewski Does `/opt/homebrew/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/exe/bundle` or `/opt/homebrew/bin/bundler` exist? Do either of them work as expected?

**Comment by bdolewski** [comment]
> @bdolewski Does `/opt/homebrew/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/exe/bundle` or `/opt/homebrew/bin/bundler` exist? Do either of them work as expected?

@MikeMcQuaid how do I test that? Should I just go to `/opt/homebrew/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/exe/` and run `./bundle` ? Unfortunately I'm not a Ruby developer, don't really know how to test those tools ;) I install Ruby on my machine because of open-source tools for iOS Developers written in Ruby.

**Comment by MikeMcQuaid** [comment]
@bdolewski Given that: are you having any problems with `ruby` beyond the `brew test` failing? If not: I think you can ignore it.

**Comment by fxcoudert** [comment]
We're at 400 bottles tonight, and bottling will continue for all software that builds. I'm closing this call for testing, thanks everyone!

**Comment by unitof** [comment]
Woah! Is there an updated list of ARM bottles, or best to just check formulae?

**Comment by iMichka** [comment]
`cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula && git grep :arm64_big_sur` will give you a quick overview.

**Comment by fxcoudert** [comment]
We're at 590 bottles two hours later, so clearly we're not going to be able to keep a list up-to-date. Looking at the formulas is the best option (@iMichka's command, but with `/opt/homebrew` as prefix if you're on Apple Silicon)

**Comment by richiksc** [comment]
@fxcoudert Hmm, it seems that the `node` bottle has mysteriously disappeared. Homebrew is installed via the untar method to `/opt/homebrew` on an M1-based Mac.

```
% brew install node
Updating Homebrew...
Warning: You are running macOS on a arm64 CPU architecture.
We do not provide support for this (yet).
Reinstall Homebrew under Rosetta 2 until we support it.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
unsupported configuration.

Error: node: no bottle available!
You can try to install from source with e.g.
  brew install --build-from-source node
Please note building from source is unsupported. You will encounter build
failures with some formulae. If you experience any issues please create pull
requests instead of asking for help on Homebrew's GitHub, Twitter or any other
official channels.
```

**Comment by fxcoudert** [comment]
@richiksc thanks for reporting, following the update the bottle has not been rebuilt immediately, I'm starting a build. Things will be a little bit out of sync in the next couple of days, as we continue to build bottles, but then we can enable ARM fully on our CI and this problem will disappear

**Comment by richiksc** [comment]
Thanks @fxcoudert, looks like the Node 15.5.0 bottle is available now. Looking forward for it to be fully integrated with the CI!

**Comment by domschl** [comment]
The arm64 bottle for mosquito is broken, `libmosquitto.1.dylib` not found:
```
brew reinstall mosquitto
==> Downloading https://homebrew.bintray.com/bottles/mosquitto-2.0.5.arm64_big_sur.bott
Already downloaded: /Users/dsc/Library/Caches/Homebrew/downloads/2e43e01e305214511e7fec05b8e1ed1d1fd52d998b0b0309275b6972e6f1d503--mosquitto-2.0.5.arm64_big_sur.bottle.tar.gz
==> Reinstalling mosquitto
==> Pouring mosquitto-2.0.5.arm64_big_sur.bottle.tar.gz
==> Caveats
mosquitto has been installed with a default configuration file.
You can make changes to the configuration by editing:
    /opt/homebrew/etc/mosquitto/mosquitto.conf

To have launchd start mosquitto now and restart at login:
  brew services start mosquitto
Or, if you don't want/need a background service you can just run:
  mosquitto -c /opt/homebrew/etc/mosquitto/mosquitto.conf
==> Summary
🍺  /opt/homebrew/Cellar/mosquitto/2.0.5: 41 files, 1.6MB
 
 $ mosquitto_sub
dyld: Library not loaded: @rpath/libmosquitto.1.dylib
  Referenced from: /opt/homebrew/bin/mosquitto_sub
  Reason: image not found
zsh: abort      mosquitto_sub
```

The library is there:

```
$ pwd
/opt/homebrew/Cellar/mosquitto/2.0.5/lib

$ ls -l
total 440
lrwxr-xr-x  1 dsc  staff      24 Jan 14 09:04 libmosquitto.1.dylib -> libmosquitto.2.0.5.dylib
-r--r--r--  1 dsc  staff  141008 Jan 14 09:04 libmosquitto.2.0.5.dylib
lrwxr-xr-x  1 dsc  staff      20 Jan 14 09:04 libmosquitto.dylib -> libmosquitto.1.dylib
lrwxr-xr-x  1 dsc  staff      26 Jan 14 09:04 libmosquittopp.1.dylib -> libmosquittopp.2.0.5.dylib
-r--r--r--  1 dsc  staff   78560 Jan 14 09:04 libmosquittopp.2.0.5.dylib
lrwxr-xr-x  1 dsc  staff      22 Jan 14 09:04 libmosquittopp.dylib -> libmosquittopp.1.dylib
drwxr-xr-x  4 dsc  staff     128 Jan 14 09:04 pkgconfig
```

There are no signing problems.

**Comment by fxcoudert** [comment]
@domschl can you trying rebuilding from source, with `-DCMAKE_INSTALL_RPATH=#{lib}` added to the `cmake` arguments in the formula?

**Comment by domschl** [comment]
@fxcoudert No change:

change to formula:
```
def install
    system "cmake", ".", *std_cmake_args, "-DCMAKE_INSTALL_RPATH=#{lib} -DWITH_WEBSOCKETS=ON"
    system "make", "install"
  end
```

output:

```
$ brew install -s mosquitto                                               1
==> Downloading https://mosquitto.org/files/source/mosquitto-2.0.5.tar.gz
Already downloaded: /Users/dsc/Library/Caches/Homebrew/downloads/2c9713316856338d478ad2203f6d54c101f7fe8294a925895c82b451f74031c7--mosquitto-2.0.5.tar.gz
==> cmake . -DCMAKE_INSTALL_RPATH=/opt/homebrew/Cellar/mosquitto/2.0.5/lib -DWITH_WEBSOCKETS=ON
==> make install
==> Caveats
mosquitto has been installed with a default configuration file.
You can make changes to the configuration by editing:
    /opt/homebrew/etc/mosquitto/mosquitto.conf

To have launchd start mosquitto now and restart at login:
  brew services start mosquitto
Or, if you don't want/need a background service you can just run:
  mosquitto -c /opt/homebrew/etc/mosquitto/mosquitto.conf
==> Summary
🍺  /opt/homebrew/Cellar/mosquitto/2.0.5: 41 files, 1.3MB, built in 5 seconds
 
 $ mosquitto_sub
dyld: Library not loaded: @rpath/libmosquitto.1.dylib
  Referenced from: /opt/homebrew/bin/mosquitto_sub
  Reason: image not found
zsh: abort      mosquitto_sub
```

**Comment by fxcoudert** [comment]
Should be:
`system "cmake", ".", *std_cmake_args, "-DCMAKE_INSTALL_RPATH=#{lib}", "-DWITH_WEBSOCKETS=ON"`

with an extra `", "` to separate the arguments

**Comment by domschl** [comment]
@fxcoudert : Thank you for your help! This works now.
